api-tests/val/nspe/pal_interfaces_ns.h

```diff
@@ -126,4 +127,18 @@ void pal_terminate_simulation(void);
  *   @return   - SUCCESS/FAILURE
 **/
 int pal_system_reset(void);
+
+/**
+ *   @brief    - Sets the custom test list buffer
+ *   @param    - custom_test_list : Custom test list buffer 
+     @return   - void
+**/
+void pal_set_custom_test_list(char *custom_test_list);
```

Comment:
May I ask you to create a function to provide a list of all the available tests (e.g. as an integer list [201,202,203,...] or a string list ["test_201;", "test_202;",...] or even a single string "test_201;test_202;...") ?
If this function would return the converted numbers, e.g:
["test_1;", "test_2;",...] my other finding about the conversion could be rejected.

Reply:
Hi @gabor-toth-arm,
The list of available tests (test_entry_list.inc) is known by the Validation Abstraction Layer, not the Platform Abstraction Layer.
Can you clarify what this function would be used for?

Reply:
We have a testrunner API, that calls the test binaries. With the new functions we are able to select which tests to be run via command line arguments, but without checking the code people will not know what tests are available. I the idea is that it would be beneficial to have a "help" argument, which prints all the tests from test_entry_list.inc. If there was an API that could provide this information to us, it could be implemented without ugly workarounds, like reading this file from our code and including random stuff from PAl or VAL to create the struct that will hold it, etc.

Reply:
I would suggest implementing a VAL function that returns a list of test IDs (integers).
Then, in the use case you described, the caller would print those test IDs in the appropriate format.
I think this is outside the original scope of this PR. I can push a new commit implementing this and possibly create a new PR later if needed.

Reply:
The suggested solution would be perfect for us. Should we create a new feature request or could you implement it under the current one ( https://github.com/ARM-software/psa-arch-tests/issues/345 ) ?